src/plugins/vis_type_table/public/utils/use/use_formatted_columns.ts

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { useMemo } from 'react';
+import { chain, findIndex } from 'lodash';
+
+import { Table } from '../../table_vis_response_handler';
+import { FormattedColumn, TableVisConfig, AggTypes } from '../../types';
+import { getFormatService } from '../../services';
+import { addPercentageColumn } from '../add_percentage_column';
+
+export const useFormattedColumnsAndRows = (table: Table, visConfig: TableVisConfig) => {
+  const { formattedColumns: columns, formattedRows: rows } = useMemo(() => {
+    const { buckets, metrics } = visConfig.dimensions;
+    let formattedRows = table.rows;
+
+    let formattedColumns = table.columns
+      .map<FormattedColumn | undefined>((col, i) => {
+        const isBucket = buckets.find(({ accessor }) => accessor === i);
+        const dimension = isBucket || metrics.find(({ accessor }) => accessor === i);
+
+        const formatter = dimension ? getFormatService().deserialize(dimension.format) : undefined;
+
+        const formattedColumn: FormattedColumn = {
+          id: col.id,
+          title: col.name,
+          formatter,
+          filterable: !!isBucket,
+        };
+
+        if (!dimension) return undefined;
```

Comment:
Let's move this early dropout a bit earlier. We don't need to fo all the `formattedColumn` and `formatter` stuff if the dimension is not set. Maybe it even helps us making `formatter` non nullable in the interface?

Reply:
done